core/trino-main/src/main/java/io/trino/metadata/FunctionMetadata.java

```diff
@@ -40,7 +41,7 @@ public FunctionMetadata(
             String description,
             FunctionKind kind)
     {
-        this(FunctionId.toFunctionId(signature), signature, nullable, argumentDefinitions, hidden, deterministic, description, kind, false);
+        this(FunctionId.toFunctionId(signature), signature, nullable, argumentDefinitions, hidden, deterministic, description, kind, false, signature.getName());
     }
 
     public FunctionMetadata(
```

Comment:
Can we modify `ParametricScalar` and `SqlWindowFunction` to use the new constructor (with resolved name being passed as an parameter) ?